src/server/scripts/Kalimdor/zone_the_barrens.cpp

```diff
@@ -69,9 +71,9 @@ class npc_beaten_corpse : public CreatureScript
     bool OnGossipHello(Player* player, Creature* creature) override
     {
         if (player->GetQuestStatus(QUEST_LOST_IN_BATTLE) == QUEST_STATUS_INCOMPLETE || player->GetQuestStatus(QUEST_LOST_IN_BATTLE) == QUEST_STATUS_COMPLETE)
-            player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, GOSSIP_CORPSE, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF+1);
+            player->ADD_GOSSIP_ITEM_DB(GOSSIP_MENU_OPTION_INSPECT_BODY, GOSSIP_OPTION_ID_BEATEN_CORPSE, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
 
-        player->SEND_GOSSIP_MENU(3557, creature->GetGUID());
+        player->SEND_GOSSIP_MENU(NPC_TEXT_1_BEATEN_CORPSE, creature->GetGUID());
```

Comment:
Oh, I see. I tried using only these lines
```cpp
            void sGossipSelect(Player* player, uint32 menuId, uint32 gossipListId) override
            {
                player->TalkedToCreature(me->GetEntry(), me->GetGUID());
            }
```
to replace the 2 existing hooks and of course it did not work. Thanks for the full example! :)